integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/util/SecondaryIndexUtil.scala

```diff
@@ -280,6 +284,40 @@ object SecondaryIndexUtil {
     }
   }
 
+  /**
+   * This method deletes the old index files or merge index file after data files merge
+   */
+  def deleteOldIndexOrMergeIndexFiles(carbonLoadModel: CarbonLoadModel,
+      validSegments: util.List[Segment],
+      indexCarbonTable: CarbonTable): Unit = {
+    // delete the index/merge index carbonFile of old data files
+    val loadModel = SecondaryIndexUtil
+      .getCarbonLoadModel(carbonLoadModel.getCarbonDataLoadSchema.getCarbonTable,
+        carbonLoadModel.getLoadMetadataDetails,
+        carbonLoadModel.getFactTimeStamp,
+        carbonLoadModel.getColumnCompressor)
+    var indexFiles: util.Map[String, String] = null
+    validSegments.asScala.foreach { segment =>
+      // for old store scenario
+      if (segment.getSegmentFileName == null) {
+        val segmentPath = CarbonTablePath.getSegmentPath(indexCarbonTable.getTablePath,
+          segment.getSegmentNo)
+        indexFiles = new SegmentIndexFileStore().getMergeOrIndexFilesFromSegment(segmentPath)
+      } else {
+        val segmentFileStore = new SegmentFileStore(indexCarbonTable.getTablePath,
+          segment.getSegmentFileName)
+        indexFiles = segmentFileStore.getIndexOrMergeFiles
+      }
+    }
+    val indexFilesToDelete = indexFiles.keySet().asScala.filter { indexFile =>
```

Comment:
I think for each segment we have to delete old indexFiles. This delete code has to be moved inside foreach loop?

Reply:
yes